coala_quickstart/green_mode/find_globs.py

```diff
@@ -0,0 +1,185 @@
+import os
+from fnmatch import fnmatch
+
+
+def find_globs_from_files(node, initials, project_dir, files, ignore_list,
+                          already_found_ext=[]):
+    """
+    Finds the globs from the files present in the section, looking first
+    for the glob '**', then the glob '**' with different filename extensions
+    present. Repeats the same process for the glob '*' on each iteration.
+    :param node:
+        The current Node object representing the file or folder we are on.
+    :param initials:
+        Constructing the name of the file by collecting names from the current
+        node and passing it along to the children.
+    :param project_dir:
+        The name of the project directory.
+    :param files:
+        The files list for the section which is appended/changed on each
+        recursion of this method.
+    :param ignore_list:
+        The list of files to ignore for the current section which is
+        constructed along as the recursion algorithm proceeds.
+    :param already_found_ext:
+        The list of extensions already covered inside the glob '**' with
+        the extension to be passed along the the children nodes to skip the
+        checks.
+    :return:
+        List of appended files and list of files to ignore for the current
+        section.
+    """
+    for child in node.children:
+        if not child.children == []:
+            if child.name == 'project_dir':
+                child_name = project_dir
+            else:
+                child_name = child.name
+            dir_name = initials + os.sep + child_name
+            dir_name = dir_name.replace(os.sep + os.sep, os.sep)
```

Comment:
manually replacing this way can break on some platforms, better to use some method in `os.path` for this

Reply:
As I recall this was forcefully introduced by earlier and its again being forcefully removed by me too